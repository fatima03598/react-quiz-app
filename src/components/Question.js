import React from 'react';

class Question extends React.Component {
    constructor(props){
        super(props)
        this.state={
            userNames: Object.values(this.props.userNames),
            playersNumber: this.props.playersNumber,
            questionsArray: [],
            score: 0,
            answerArray: [],
            userScores: {

            },
        }
    };

    handleChange = (event) => {
        const {value} = event.target;
        const user = event.target.id;
        if((value === atob(this.props.questionContent.correct_answer)) && (this.state.userScores[user] == 0)) { 
            this.props.totalScore(1, user);
            this.setState( { userScores: { ...this.state.userScores, [user]: 1}} )
        }else if((value !== atob(this.props.questionContent.correct_answer)) && (this.state.userScores[user] == 1)){
            this.props.totalScore(-1, user);
            this.setState( { userScores: { ...this.state.userScores, [user]: 0}} )
        };
        console.log(this.state.userScores)
    }

    componentDidMount(){
        this.shuffleArray();
        this.state.userNames.map(user => {
            this.setState({ userScores: { ...this.state.userScores, [user]: 0}})})
    };

    shuffleArray = () => {
        let answers = new Array;
        answers.push(atob(this.props.questionContent.correct_answer));
        this.props.questionContent.incorrect_answers.forEach((answer) => answers.push(atob(answer)));
        answers.sort(() => Math.random() -0.5);
        this.setState({ answerArray: answers })
    }
    
    // if checked button === correct_answer, setState score + 1


    render(){
        console.log(this.state);

    return(

        <div className="Question">
            <h2>Question {this.props.id + 1}</h2>
            <h3>{atob(this.props.questionContent.question)}</h3>
            {this.state.userNames.map(user => {
                return(
                <form>
                <h4>{user}</h4>
                {this.state.answerArray.map(answer => {
                    return(
                        <div>
                                <input 
                                    type="radio" 
                                    id={user} 
                                    name={`answer_${this.props.id}`} 
                                    value={answer}
                                    onChange={this.handleChange}
                                    />
                                <label for={answer}> {answer} </label>
                            
                        </div>
                    )
                })}
                </form>
            )})}

        </div>

    )   
    }
}
export default Question;



