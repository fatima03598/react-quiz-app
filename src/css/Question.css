.question{
    display: flex;
    flex-direction: column;
}

.question-container{
    display: flex;
    flex-direction: column;
}
.radio-container{
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}
.radio-buttons{
    display: flex;
    flex-direction: row;
}