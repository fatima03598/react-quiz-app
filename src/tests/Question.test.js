import React from 'react'
import { BrowserRouter, Link } from 'react-router-dom';
import { shallow, mount } from 'enzyme'
import Question from '../components/Question'


describe('<Question />', () => {

    it('should render main Question div', () => {
        const wrapper = shallow(<Question key={1} id={1} questionContent={{category:"General Knowledge",type:"multiple",difficulty:"easy",question:"What is the largest organ of the human body?", correct_answer:"Skin",incorrect_answers:["Heart","Intestine","Liver"]}} />)
        expect(wrapper.exists('.Question')).toEqual(true);
    } )
})







//npm test -- --coverage --watchAll=false  
    