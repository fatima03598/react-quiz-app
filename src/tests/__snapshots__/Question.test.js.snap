// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question matches the snapshot 1`] = `
<div
  className="question-container"
  id="category-9"
>
  <div
    className="question-inner-container"
  >
    <h4>
      Question 
      1
    </h4>
    <h2>
      In the video-game franchise Kingdom Hearts, the main protagonist, carries a weapon with what shape?
    </h2>
    <div
      className="radio-container"
    >
      <div
        className="answer-list"
      >
        <h3
          className="player-name"
        >
          Rorie
        </h3>
        <form>
          <div
            className="single-answer"
          >
            <input
              className="radio-button-answer"
              id="Rorie"
              name="answer_0"
              onChange={[Function]}
              type="radio"
              value="Pen"
            />
            <label
              for="Pen"
            >
               
              Pen
               
            </label>
          </div>
          <div
            className="single-answer"
          >
            <input
              className="radio-button-answer"
              id="Rorie"
              name="answer_0"
              onChange={[Function]}
              type="radio"
              value="Cellphone"
            />
            <label
              for="Cellphone"
            >
               
              Cellphone
               
            </label>
          </div>
          <div
            className="single-answer"
          >
            <input
              className="radio-button-answer"
              id="Rorie"
              name="answer_0"
              onChange={[Function]}
              type="radio"
              value="Key"
            />
            <label
              for="Key"
            >
               
              Key
               
            </label>
          </div>
          <div
            className="single-answer"
          >
            <input
              className="radio-button-answer"
              id="Rorie"
              name="answer_0"
              onChange={[Function]}
              type="radio"
              value="Sword"
            />
            <label
              for="Sword"
            >
               
              Sword
               
            </label>
          </div>
        </form>
      </div>
    </div>
    <div
      className="next-question-button-container"
    >
      <a
        href="/nextquestion/0"
        onClick={[Function]}
      >
        <button
          className="next-question-button"
          type="submit"
        >
          Check Answer!
        </button>
      </a>
    </div>
  </div>
</div>
`;
